Plugins/NativeXr/Source/NativeXr.cpp

```diff
@@ -46,30 +46,19 @@ namespace
 
     std::array<float, 16> CreateProjectionMatrix(const xr::System::Session::Frame::View& view)
     {
-        const float n{view.DepthNearZ};
-        const float f{view.DepthFarZ};
-
-        const float r{std::tanf(view.FieldOfView.AngleRight) * n};
-        const float l{std::tanf(view.FieldOfView.AngleLeft) * n};
-        const float t{std::tanf(view.FieldOfView.AngleUp) * n};
-        const float b{std::tanf(view.FieldOfView.AngleDown) * n};
-
-        // Angles for FieldOfView respect the viewport ratio
-        // but tangent is not a linear function and ratio of values(l,r,b,t) computed
-        // in CreateProjectionMatrix do not respect that ratio
-        // so, here, a ratio after tangent is computed
-        // and a second derivative ratio computed to compensate
-        // the aspect ratio delta after tangent calls
-        const float aspectRatio = static_cast<float>(view.ColorTextureSize.Width) / static_cast<float>(view.ColorTextureSize.Height);
-        const float deltax = (r - l);
-        const float deltay = (t - b);
-        const float afterTangentAspectRatio = deltax / deltay;
-        const float compensationRatio = afterTangentAspectRatio / aspectRatio;
-        const float tc{std::tanf(view.FieldOfView.AngleUp * compensationRatio) * n};
-        const float bc{std::tanf(view.FieldOfView.AngleDown * compensationRatio) * n};
+        // Calculate total vertical and horizontal FoV and the aspect ratio.
+        const float verticalFoV{abs(view.FieldOfView.AngleUp) + abs(view.FieldOfView.AngleDown)};
+        const float horizontalFoV{abs(view.FieldOfView.AngleRight) + abs(view.FieldOfView.AngleLeft)};
```

Comment:
This solution assumes symmetric FOVs.  Do we have cases in our use cases on OpenXR with asymmetric FOVs?

@syntheticmagus @rgerd @chrisfromwork

Reply:
I'm not sure that we can assume FOVs are symmetrical. I can test this out on a HoloLens tomorrow

Reply:
Was able to get to this today. HoloLens FOV's aren't symmetrical. we should sync on this offline tomorrow.

Reply:
Sounds good for now I am going to fork the behavior and keep the old implementation for asymmetric FoVs.  We should investigate if whether the old calculation was correct for these cases.

Reply:
After a lengthy discussion, we're going to instead push projection matrix creation down to platform specific implementations.